StaffControl/Application/Services/ReceptonistService.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AutoMapper;
+using StaffControl.Application.DTOs.ReceptionistDTOs;
+using StaffControl.Application.Interfaces;
+using StaffControl.Domain.Entities;
+using StaffControl.Domain.Interfaces;
+using StaffControl.Infrastructure.Interfaces;
+
+namespace StaffControl.Application.Services
+{
+    public class ReceptonistService : IReceptionistService
+    {
+        private readonly IReceptionistRepository _receptionistRepository;
+        private readonly IMapper _mapper;
+        private readonly ILogService _logger;
+
+        public ReceptonistService(IReceptionistRepository receptionistRepository, IMapper mapper, ILogService logger)
+        {
+            _receptionistRepository = receptionistRepository;
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<Receptionist>> GetAllAsync()
+        {
+            return await _receptionistRepository.GetAllAsync();
+        }
+
+        public async Task<Receptionist?> GetByIdAsync(Guid id)
+        {
+            var receptionist = await _receptionistRepository.GetByIdAsync(id);
+
+            if (receptionist == null)
+            {
+                _logger.LogWarning($"Receptionist with ID {id} was not found.");
+            }
+
+            return receptionist; ;
+        }
+
+        public async Task AddAsync(CreateReceptionistDto createReceptionistDto)
+        {
+            var receptionist = _mapper.Map<Receptionist>(createReceptionistDto);
+
+            await _receptionistRepository.AddAsync(receptionist);
+            _logger.LogInfo($"Receptionist with ID {receptionist.Id} was added: {receptionist.FirstName} {receptionist.LastName}.");
+
+        }
+
+        public async Task<bool> DeleteAsync(Guid id)
+        {
+            var receptionist = await _receptionistRepository.GetByIdAsync(id);
```

Comment:
Unnecessary call for the db. You're calling get by id twice: once here and the second one in DeleteAsync. It's enough to call it once in delete. And you can have all you logic depending on what is returned `_receptionistRepository.DeleteAsync(id);` , though now this value is ignored hic isn't good